continuous_integration/recipe/meta.yaml

```diff
@@ -22,9 +22,12 @@ build:
 
 requirements:
   build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - {{ compiler('c') }}
     - {{ compiler('rust') }}
     - setuptools-rust >=1.5.2
+    - libprotobuf =3
```

Comment:
Curious what prompted adding `libprotobuf` to `build`?

Reply:
I would've thought we needed it here before this PR, as adding `datafusion-substrait` to our Rust dependencies once required `protoc` for compilation (cc @jdye64), but I suppose it was getting picked up transitively up until now?

I recall getting failures building the `linux-aarch64` packages due to missing the `protoc`, can double check that this is the case; if so, I suppose we could add a selector here to only pick this up for `linux-aarch64`, but thought it made sense to specify explicitly for all builds.

Reply:
That should already be possible with the dependency in `host`. No need to have it in `build` also

Would think of `libprotobuf` more like `python` and less like `cmake` as it is something we want in the environment installed into as we may want to link to it

Reply:
Looks like builds are failing without `libprotobuf` specified as a build-time requirement:

```
  error: failed to run custom build command for `substrait v0.7.5`
  note: To improve backtraces for build dependencies, set the CARGO_PROFILE_RELEASE_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

  Caused by:
    process didn't exit successfully: `/usr/share/miniconda3/envs/test/conda-bld/dask-sql_1684428974210/work/dask_planner/target/release/build/substrait-79784529fc7e3b5c/build-script-build` (exit status: 1)
    --- stdout
    cargo:rerun-if-env-changed=FORCE_REBUILD
    cargo:rerun-if-changed=substrait
    cargo:rerun-if-changed=substrait/text/simple_extensions_schema.yaml
    cargo:rerun-if-changed=substrait/proto/substrait/algebra.proto
    cargo:rerun-if-changed=substrait/proto/substrait/type_expressions.proto
    cargo:rerun-if-changed=substrait/proto/substrait/extensions/extensions.proto
    cargo:rerun-if-changed=substrait/proto/substrait/plan.proto
    cargo:rerun-if-changed=substrait/proto/substrait/extended_expression.proto
    cargo:rerun-if-changed=substrait/proto/substrait/capabilities.proto
    cargo:rerun-if-changed=substrait/proto/substrait/type.proto
    cargo:rerun-if-changed=substrait/proto/substrait/function.proto
    cargo:rerun-if-changed=substrait/proto/substrait/parameterized_types.proto

    --- stderr
    Error: Custom { kind: Other, error: "protoc failed: /usr/share/miniconda3/envs/test/conda-bld/dask-sql_1684428974210/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin/protoc: 1: Syntax error: \"(\" unexpected\n" }
  warning: build failed, waiting for other jobs to finish...
```

cc @jdye64 not sure if this looks familiar to you?

Reply:
Added new review comments below related to this